test/test_optim.py

```diff
@@ -4026,30 +4035,40 @@ def _test_averaged_model(self, net_device, swa_device):
             self.assertTrue(p.device == net_device)
         self.assertTrue(averaged_dnn.n_averaged.device == swa_device)
 
-    def test_averaged_model_all_devices(self):
+    @parametrize("ema", [True, False])
+    def test_averaged_model_all_devices(self, ema):
         cpu = torch.device("cpu")
-        self._test_averaged_model(cpu, cpu)
+        self._test_averaged_model(cpu, cpu, ema)
         if torch.cuda.is_available():
             cuda = torch.device(0)
-            self._test_averaged_model(cuda, cpu)
-            self._test_averaged_model(cpu, cuda)
-            self._test_averaged_model(cuda, cuda)
+            self._test_averaged_model(cuda, cpu, ema)
+            self._test_averaged_model(cpu, cuda, ema)
+            self._test_averaged_model(cuda, cuda, ema)
 
-    def test_averaged_model_mixed_device(self):
+    @parametrize("ema", [True, False])
+    def test_averaged_model_mixed_device(self, ema):
```

Comment:
Could this and _test_averaged_model  be refactored to call a common helper? There seems to be a bit of duplication.

Reply:
Done